src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1128,7 +1128,10 @@ QgsRectangle QgsOgrProvider::extent() const
     // get the extent_ (envelope) of the layer
     QgsDebugMsgLevel( QStringLiteral( "Starting get extent" ), 3 );
 
-    if ( mForceRecomputeExtent && mValid && mGDALDriverName == QLatin1String( "GPKG" ) && mOgrOrigLayer )
+    if ( mForceRecomputeExtent && mValid &&
+         ( mGDALDriverName == QLatin1String( "GPKG" ) ||
+           ( mGDALDriverName == QLatin1String( "ESRI Shapefile" ) && mWriteAccess) ) &&
```

Comment:
why the mWriteAccess test is just for shapefile ? It should also apply to GPKG

Reply:
This is because when I tried bug_extent_of_shapefile.py without mWriteAccess test, only ESRI Shapefile had following error:

ERROR 6: RecomputeExtent : unsupported operation on a read-only datasource.

I didn't add the code for GPKG because it didn't have the error originally.
I'll add the test for GPKG in the next commit.